File: 5-svr/svr.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
 
dataset = pd.read_csv("Position_Salaries.csv")
x = dataset.iloc[:,1:2].values
y = dataset.iloc[:,2].values


from sklearn.preprocessing import StandardScaler
sc_x = StandardScaler()
sc_y = StandardScaler()
x = sc_x.fit_transform(x)
y = sc_y.fit_transform(y.reshape(-1,1))


from sklearn.svm import SVR
regression = SVR(kernel = "rbf")
regression.fit(x,y)

seis_cinco = np.array([[6.5]]) 

seis = sc_x.transform(seis_cinco)

y_pred = regression.predict(seis)
y_pred = sc_y.inverse_transform(y_pred)

x_grid = np.arange(min(x),max(x),0.1)
x_grid = x_grid.reshape(len(x_grid),1)
y_grid = regression.predict(x_grid)

plt.scatter(sc_x.inverse_transform(x),sc_y.inverse_transform(y),color = "red")
plt.plot(sc_x.inverse_transform(x_grid),sc_y.inverse_transform(y_grid),color = "blue")
plt.title("Modelo de regresion (SVR)")
plt.xlabel("Posicion del empleado")
plt.ylabel("Sueldo en ($)")
plt.show()

File: 5-svr/svr.R
dataset = read.csv("Position_Salaries.csv")
dataset = dataset[,2:3]

install.packages("e1071")
library(e1071)

regression = svm(formula = Salary ~ .,
                 data = dataset,
                 type = "eps-regression",
                 kernel = "radial")

y_pred = predict(regression, newdata = data.frame(Level = 6.5))


library(ggplot2)
x_grid = seq(min(dataset$Level),max(dataset$Level), 0.1)
dataset2 = data.frame(Level = x_grid,Level2 = x_grid^2,Level3 = x_grid^3)
ggplot() +
  geom_point(aes(x =  dataset$Level, y = dataset$Salary),
             color = "red") +
  geom_line(aes(x = dataset$Level,y = predict(regression, newdata = dataset)),
            color = "blue") +
  ggtitle("Prediccion del sueldo en funcion del nivel") +
  xlab("Nivel del empleado") +
  ylab("Sueldo en $")